src/services/joinWaitListService.ts

```diff
@@ -0,0 +1,34 @@
+import CustomError from "../middlewares/customError"
+import { getProspect } from "../queries/prospectQuery"
+import { addProspectToTheWaitList } from "../queries/waitListQuery"
+import { io, room } from "../server"
+import { USER_NOT_FOUND } from "../utils/errorCodes"
+import { JoinWaitListDto } from "../utils/types"
+
+const joinWaitListService = async (joinWaitListData: JoinWaitListDto) => {
+  const recipient = room[0]
+  const email = joinWaitListData.responses.email
+
+  const prospect = await getProspect(email)
+
+  if (prospect && !prospect.isOnWaitList) {
+    const addedProspect = await addProspectToTheWaitList(email)
+
+    io.to(recipient).emit("joinedTheWaitList", { email })
+
+    return addedProspect
+  } else {
```

Comment:
it's not necessary to add an else when you have a return in the previous if condition